roles/openshift_manageiq/tasks/main.yaml

```diff
@@ -0,0 +1,85 @@
+---
+- name: Copy Configuration to temporary conf
+  command: >
+    cp {{ openshift.common.config_base }}/master/admin.kubeconfig /tmp/manageiq_admin.kubeconfig
+
+- name: Add Managment Infrastructure project
+  command: > 
+    {{ openshift.common.admin_binary }} new-project
+    management-infra
+    --description="Management Infrastructure"
+    --config=/tmp/manageiq_admin.kubeconfig
+  register: osmiq_create_mi_project
+  failed_when: "'already exists' not in osmiq_create_mi_project.stderr and osmiq_create_mi_project.rc != 0"
+  changed_when: false
+
+- name: Create Service Account
+  shell: >
+    echo {{ manageiq_service_account | to_json | quote }} | 
+    {{ openshift.common.client_binary }} create 
+    -n management-infra 
+    --config=/tmp/manageiq_admin.kubeconfig
+    -f -
+  register: osmiq_create_service_account
+  failed_when: "'already exists' not in osmiq_create_service_account.stderr and osmiq_create_service_account.rc != 0"
+  changed_when: false
+
+- name: Create Cluster Role
+  shell: >
+    echo {{ manageiq_cluster_role | to_json | quote }} |
+    {{ openshift.common.client_binary }} create 
+    --config=/tmp/manageiq_admin.kubeconfig
+    -f -
+  register: osmiq_create_cluster_role
+  failed_when: "'already exists' not in osmiq_create_cluster_role.stderr and osmiq_create_cluster_role.rc != 0"
+  changed_when: false
+
+- name: Add role to admin
+  command: >
+    {{ openshift.common.admin_binary }} policy
+    add-role-to-user 
+    -n management-infra
+    admin 
+    -z managment-admin
+    --config=/tmp/manageiq_admin.kubeconfig
+  register: osmiq_add_role_admin
+  failed_when: "'already exists' not in osmiq_add_role_admin.stderr and osmiq_add_role_admin.rc != 0"
+  changed_when: false
+
+- name: Add role to management infra admin
+  command: >
+    {{ openshift.common.admin_binary }} policy
+    add-role-to-user 
+    -n management-infra
+    management-infra-admin 
+    -z managment-admin
+    --config=/tmp/manageiq_admin.kubeconfig
+  register: osmiq_add_role_miadmin
+  failed_when: "'already exists' not in osmiq_add_role_miadmin.stderr and osmiq_add_role_miadmin.rc != 0"
+  changed_when: false
```

Comment:
Since a lot of these are very similar, it may make sense to loop through them instead of having many individual tasks. Obviously since 2 of them require the use of the shell module it can't be done for all admin_binary tasks.

in roles/openshift_manageiq/vars/main.yml:
```
manage_iq_tasks:
- policy add-role-to-user admin -z management-admin -n management-infra
- policy add-role-to-user management-infra-admin -z management-admin -n management-infra
- policy add-cluster-role-to-user cluster-reader system:serviceaccount:management-infra:management-admin
- policy add-scc-to-user privileged system:serviceaccount:management-infra:management-admin
```
in roles/openshift_manageiq/tasks/main.yaml:
```
- name: Configure role/user permissions
  command: >
    {{ openshift.common.admin_binary }} item --config=/tmp/manageiq_admin.kubeconfig
  register: osmiq_perm_task
  failed_when: "'already exists' not in osmiq_perm_task.stderr and osmiq_perm_task.rc != 0"
  changed_when: osmiq_perm_task.rc != 0
```